File: fastapi/src/middlewares/_public_paths.py
_PUBLIC_PREFIXES: tuple[str, ...] = (
    "/docs",
    "/redoc",
    "/openapi.json",
    # projx-anchor: public-prefixes
)

_AUTHN_ONLY_PREFIXES: tuple[str, ...] = (
    # projx-anchor: authn-only-prefixes
)
_PUBLIC_EXACT: tuple[str, ...] = (
    "/api/",
    "/api/health",
    "/api/health/live",
    "/api/health/ready",
    # projx-anchor: public-exact
)


def is_public_path(path: str) -> bool:
    return path in _PUBLIC_EXACT or path.startswith(_PUBLIC_PREFIXES)


def is_authn_only_path(path: str) -> bool:
    return any(p and path.startswith(p) for p in _AUTHN_ONLY_PREFIXES)

File: express/src/middlewares/public-paths.ts
const PUBLIC_PREFIXES = [
  '/docs',
  '/documentation',
  '/openapi.json',
  '/swagger',
];
const PUBLIC_EXACT = new Set([
  '/api/',
  '/api/health',
  '/api/health/live',
  '/api/health/ready',
]);
const AUTHN_ONLY_PREFIXES: string[] = [];

export function isPublicPath(path: string): boolean {
  return (
    PUBLIC_EXACT.has(path) || PUBLIC_PREFIXES.some((p) => path.startsWith(p))
  );
}

export function isAuthnOnlyPath(path: string): boolean {
  return AUTHN_ONLY_PREFIXES.some((p) => path.startsWith(p));
}
